### ISSUE TITLE
Menu item size e.g. X of X no longer reported by screen readers

### ISSUE THREAD

**Issue by pbossley** [wet-boew/wet-boew#8081]
Verified in working example site: https://wet-boew.github.io/v4.0-ci/index-en.html
Firefox 54.0.1 and NVDA 2017.2, JAWS 18.
When navigating menus with screen readers the set size and position in the menu is no longer reported by screen readers and now just announce 1 of 1 instead of properly reporting position in the set. 
This may be caused by ARIA set size not being defined.

**Comment by LaurentGoderre** [comment]
@shawnthompson this is caused by b9c37a90ad4a4a6c85381638011ec0d92e1cd236

**Comment by pbossley** [comment]
Interestingly, the ARIA Authoring Practices example does expose the size to screen readers. https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html
I'm not sure why that one is working properly as it doesn't appear all that different from what is being done with WET and it also doesn't seem to use set size or position in set... In any case, I would argue that this issue should in fact be solved as users should be given the correct information to orient them to the position in the overall set.

**Comment by pbossley** [comment]
If role=none is applied to the <li> items throughout, screen readers will go back to announcing position properly. Even without the use of ARIA set size. Is there any way that can be done?

**Comment by shawnthompson** [comment]
@pbossley I can work on this with you.

Thanks for your help in this.

**Comment by pbossley** [comment]
Great. Please let me know if you need any further information and I'll try to help.
Pete Bossley
Director, Digital Accessibility Center
The Ohio State University | Office of University Compliance and Integrity

**Comment by shawnthompson** [comment]
@pbossley Here is the code, which `<li>` should the `role="none"` go on?

```html
<nav role="navigation" typeof="SiteNavigationElement" id="sm-pnl" class="sm-pnl wb-menu wb-menu-inited wb-init wb-navcurr-inited">
	<h3>Topics menu</h3>
	<ul class="list-unstyled mb-menu" role="menu">
		<li>
			<details>
			<summary class="mb-item" aria-haspopup="true" tabindex="0"><span role="menuitem">WET project</span></summary>
			<ul class="list-unstyled mb-sm" role="menu" aria-expanded="false" aria-hidden="true">
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/v4.0-ci/index-en.html#about">About the Web Experience Toolkit</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/v4.0-ci/index-en.html#benefits">Benefits</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/ref/accolades-en.html">Awards, articles and videos</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/events-en.html">Events</a></li>
				<li><a role="menuitem" tabindex="-1" href="https://github.com/wet-boew/wet-boew/">Main project page</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/v4.0-ci/License-en.html">Terms and conditions</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/versions/rdmp-en.html">Roadmap of past and future releases</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/versions/index-en.html">Version history</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://www.tbs-sct.gc.ca/ws-nw/index-eng.asp">About the Web Standards</a></li>
			</ul>
			</details>
		</li>
		<li>
			<details>
			<summary class="mb-item" aria-haspopup="true" tabindex="-1"><span role="menuitem">Implement WET</span></summary>
			<ul class="list-unstyled mb-sm" role="menu" aria-expanded="false" aria-hidden="true">
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/start-en.html">Getting started with WET</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/versions/dwnld-en.html">Download WET</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/v4.0-ci/demos/index-en.html">Working examples</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/index-en.html">Documentation</a></li>
			</ul>
			</details>
		</li>
		<li>
			<details>
			<summary class="mb-item" aria-haspopup="true" tabindex="-1"><span role="menuitem">Contribute to WET</span></summary>
			<ul class="list-unstyled mb-sm" role="menu" aria-expanded="false" aria-hidden="true">
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/bugs-en.html">Filing a bug or an issue</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/pull-en.html">Creating a pull request</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/pull-en.html#test">Testing your code</a></li>
				<li><a role="menuitem" tabindex="-1" href="http://wet-boew.github.io/wet-boew/docs/start-en.html#develop">Developing for WET</a></li>
			</ul>
			</details>
		</li>
	</ul>
</nav>
```

**Comment by shawnthompson** [comment]
The one that wraps the `<details>` or the one that wraps the `<a roll="menuitem">`?

**Comment by shawnthompson** [comment]
Scratch that, I put in the wrong menu code... I will update it shortly. Sorry

**Comment by pbossley** [comment]
The role=none has to be on every list item, including the top-level menus. See the example I listed above from W3c for an example; that one is instantiating the count properly for the accessibility API.

**Comment by shawnthompson** [comment]
I just notice that the example page you gave does not pass W3C validation of the HTML code.

https://validator.w3.org/nu/?doc=https%3A%2F%2Fwww.w3.org%2FTR%2Fwai-aria-practices-1.1%2Fexamples%2Fmenubar%2Fmenubar-1%2Fmenubar-1.html

That is same reason why we removed the aria in the first place.

**Comment by pbossley** [comment]
I haven’t tested for sure, but I would assume role=”none presentation” would work as well. That would probably pass validation as that is the recommendation.

**Comment by shawnthompson** [comment]
I created a working example. For testing.

https://shawnthompson.github.io/wet-boew-aria-menu/index-en.html

**Comment by pbossley** [comment]
Working great in Firefox 54.0.1 with NVDA and JAWS 18.
In IE 11, the size and position in menu is not being reported, but that is no worse than the current situation since IE apparrently doesn't expose the right info to the A11y API. 
I'll try to test on mobile later this afternoon.

**Comment by shawnthompson** [comment]
Still does not validate though

**Comment by pbossley** [comment]
Is that a core requirement? I see this menu position issue as being a pretty big functional gap.

**Comment by shawnthompson** [comment]
It is for us. Is there anything else we can do?

**Comment by shawnthompson** [comment]
@sideshowbarker (I hope you don't mind me mentioning you) talks about it in this issue.

https://github.com/w3c/aria-practices/issues/410

**Comment by pbossley** [comment]
Role="presentation" may work, although the future of that is uncertain. Other than that, and the already-removed aria set size, I'm not aware of any other techniques.

**Comment by shawnthompson** [comment]
We removed the aria set size because it was giving us validation issues and no one knew enough about it to fix it.

I'll change it to presentation and you can see how well it works. 

Thanks again for your help.

**Comment by shawnthompson** [comment]
Ok now it validates with just presentation.

https://shawnthompson.github.io/wet-boew-aria-menu/index-en.html

@pbossley how does it read?

**Comment by pbossley** [comment]
The reporting of position and size of the menu is now much improved.

**Comment by shawnthompson** [comment]
Improved but not perfect? Should we put this in place or would you like to try other options?

**Comment by pbossley** [comment]
It works very well, as well as it can with the current landscape of user agent support. I would say if it could be put into place it would be a worthwhile improvement.

Your working on this is much appreciated; I point to WET often when working with folks on accessibility on my campus.

Thank you,

**Comment by shawnthompson** [comment]
@pbossley can you email me at shawn.thompson@servicecanada.gc.ca, I'd like to ask you a few things?

**Comment by pbossley** [comment]
Do we think this can make it into the next release?

**Comment by shawnthompson** [comment]
We will do our best!

**Comment by shawnthompson** [comment]
@ognil We are making this change today and it should be in the next release.

Are you in agreement that this change will be suffice?

https://shawnthompson.github.io/wet-boew-aria-menu/index-en.html

Thanks!

**Comment by ognil** [comment]
Thank you Shawn,

 

It works on Mozilla Firefox ONLY!

 

Sincerely,

George

**Comment by juleskuehn** [comment]
No longer an issue for me with NVDA on Firefox 60 and IE11, and also with ChromeVox. I believe this issue can be closed.

**Comment by nricherbrule** [comment]
@bsouster This issue seems to be resolved.
